### ISSUE TITLE
arrange(desc(col)) breaks sql building when followed by select()

### ISSUE THREAD

**Issue by misea** [tidyverse/dbplyr#1403]
This query is broken

```
lf <- lazy_frame(x = 1, y = 2)
lf %>% arrange(desc(x)) %>% select(x)

Error in `purrr::map_chr()`:
ℹ In index: 1.
Caused by error in `as_string()`:
! Can't convert a call to a string.
Run `rlang::last_trace()` to see where the error occurred.
```

Weirdly, this works

```
> lf <- lazy_frame(x = 1)
> lf %>% arrange(desc(x)) %>% select(x)
<SQL>
SELECT *
FROM `df`
ORDER BY `x` DESC
```

as does this

```
> lf <- lazy_frame(x = 1, y = 2)
> lf %>% arrange(x) %>% select(x)
<SQL>
SELECT `x`
FROM `df`
ORDER BY `x`
```

**Comment by misea** [comment]
I'm guessing this might have something to do with the fact that ORDER BY is not legal in an intermediate query, but error doesn't give much indication that's the case.

**Comment by mgirlich** [comment]
This got fixed in 2.4.0 😄 

> `select()` can once again be used after `arrange(desc(x))` (@ejneer, #1240).

**Comment by misea** [comment]
Thank you, I was on 2.3.3
